subsys/bluetooth/mesh/lightness_srv.c

```diff
@@ -177,7 +177,8 @@ void lightness_srv_disable_control(struct bt_mesh_lightness_srv *srv)
 void lightness_srv_change_lvl(struct bt_mesh_lightness_srv *srv,
 			      struct bt_mesh_msg_ctx *ctx,
 			      struct bt_mesh_lightness_set *set,
-			      struct bt_mesh_lightness_status *status)
+			      struct bt_mesh_lightness_status *status,
+			      bool publish)
```

Comment:
I'm not a big fan of these "and also, call this function for me" flags, but it works, and it's an internal API, so I won't put up too much resistance :slightly_smiling_face:

Reply:
I also don't really like this. Calling `bt_mesh_lightness_srv_pub` instead of publishing from this function won't work, since it has to be done with the actual lightness representation. Adding another function like `lightness_srv_change_lvl_no_publish` is kind of the same as adding `publish` argument...

Reply:
Good point. Perhaps we'll be able to improve this later, but it's not really the end of the world.